src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeListCasePersonalisation.java

```diff
@@ -59,22 +75,53 @@ public String getReferenceId(Long caseId) {
-            .put("hearingTime", dateTimeExtractor.extractHearingTime(hearingDetailsFinder.getHearingDateTime(asylumCase)))
-            .put("hearingCentreAddress", hearingDetailsFinder.getHearingCentreLocation(asylumCase));
+        Builder<String, String> listCaseFields;
+
+        if (isAcceleratedDetainedAppeal(asylumCase)) {
+            final Direction direction =
+                directionFinder
+                    .findFirst(asylumCase, DirectionTag.ADA_LIST_CASE)
+                    .orElseThrow(() -> new IllegalStateException("LR List ADA Case direction is not present"));
+
+            listCaseFields = ImmutableMap
+                .<String, String>builder()
+                .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+                .put("appealReferenceNumber", asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("ariaListingReference", asylumCase.read(ARIA_LISTING_REFERENCE, String.class).orElse(""))
+                .put("legalRepReferenceNumber", asylumCase.read(LEGAL_REP_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("appellantGivenNames", asylumCase.read(APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+                .put("appellantFamilyName", asylumCase.read(APPELLANT_FAMILY_NAME, String.class).orElse(""))
+                .put("explanation", direction.getExplanation())
+                .put("linkToOnlineService", iaExUiFrontendUrl)
+                .put("hearingCentreAddress", hearingDetailsFinder.getHearingCentreLocation(asylumCase));
+        } else {
+            listCaseFields = ImmutableMap
+                .<String, String>builder()
+                .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+                .put("appealReferenceNumber", asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("ariaListingReference", asylumCase.read(ARIA_LISTING_REFERENCE, String.class).orElse(""))
+                .put("legalRepReferenceNumber", asylumCase.read(LEGAL_REP_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("appellantGivenNames", asylumCase.read(APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+                .put("appellantFamilyName", asylumCase.read(APPELLANT_FAMILY_NAME, String.class).orElse(""))
+                .put("linkToOnlineService", iaExUiFrontendUrl)
+                .put("hearingDate", dateTimeExtractor.extractHearingDate(hearingDetailsFinder.getHearingDateTime(asylumCase)))
+                .put("hearingTime", dateTimeExtractor.extractHearingTime(hearingDetailsFinder.getHearingDateTime(asylumCase)))
+                .put("hearingCentreAddress", hearingDetailsFinder.getHearingCentreLocation(asylumCase));
+        }
 
         PersonalisationProvider.buildHearingRequirementsFields(asylumCase, listCaseFields);
 
         return listCaseFields.build();
 
     }
+
+    private boolean isAcceleratedDetainedAppeal(AsylumCase asylumCase) {
```

Comment:
Is there a place we can move this so that we can reuse this? Few notifications coming up also require this.

Reply:
Added AsylumCaseUtils